### PR TITLE
ISLANDORA-2070 Fix losing query params

### PR DESCRIPTION
**Pull request by lucasvanschaik**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2070

# What does this Pull Request do?

When the user selects another child, the query parameters of the
original URL are lost. This prevends the solr search navigation from
working correctly. This fix adds the query parameters to the links of
the children.

# What's new?

This PR adds the query parameters to the links to the compound children in the template file and for the JAIL block.

# How should this be tested?

* Enable search navigation block (admin/islandora/search/islandora_solr/settings)
* Add "islandora search navigation" block to the page (admin/structure/block) and configure it if necessary
* Do a search that has compound objects in it's results
* Click a result
* Navigate to the next and previous results with the navigation links
* When on a compound object, click another child of this compound object
* The navigation links should be visible and working
* Repeat above steps for the JAIL block

# Additional Notes:

If the islandora-compound-prev-next.tpl.php is overwritten in your theme, make sure you do the exact same change there too.

# Interested parties

@Islandora/7-x-1-x-committers @whikloj @DiegoPino

### CODE REVIEW

**Comment by DiegoPino** [comment]
Looks good and straightforward. Will wait for travis-ci to end, then 24hrs wait period and merge. Thanks

**Review by DiegoPino — APPROVED** [review]
Simple fix. Looks fine

**Comment by DiegoPino** [comment]
@lucasvanschaik since this is a bug fix and we are in Islandora 7.x-1.10 release mode we have two options here. 
1.- merge this while on release process. In that case, we need a second pull request with the same changes against the 7.x-1.10 branch, that way the fix becomes part of the release.
2.- Second option would be to wait until release process finishes (Oct 31). 

Since the bug itself is not a blocker, that is also possible.

It is up to you. I would go for 1.- 😄  Feel free to ping me

**Comment by lucasvanschaik** [comment]
@DiegoPino I have made a second pull request on the 7.x-1.10 branch. Thanks for the help!

**Comment by DiegoPino** [comment]
@lucasvanschaik good! Good release pull (clean). Still a few hours before the 24hrs limit. Will add a @Islandora/7-x-1-x-committers mention here in case someone wants to comment. Merging in 5 hours. 🎈

**Comment by whikloj** [comment]
As mentioned [here](https://github.com/Islandora/islandora_solution_pack_compound/pull/133#issuecomment-332909855) the JAIL block should also be updated.

**Comment by lucasvanschaik** [comment]
I have added the code for the JAIL block also. And I've tested it also successfully.
@whikloj thanks for your reminder of the JAIL block, since we don't use it, I did overlook it.
In your suggestions you mention https://github.com/Islandora/islandora_solution_pack_compound/blob/7.x/includes/blocks.inc#L75 , but I think that href key is not needed.

**Review by whikloj — APPROVED** [review]

**Pull request merged**
